src/Compilers/CSharp/Test/Syntax/Syntax/SyntaxNodeTests.cs

```diff
@@ -14,6 +14,49 @@ namespace Microsoft.CodeAnalysis.CSharp.UnitTests
 {
     public class SyntaxNodeTests
     {
+        [Fact]
+        [WorkItem(565382, "https://developercommunity.visualstudio.com/content/problem/565382/compiling-causes-a-stack-overflow-error.html")]
+        public void TestLargeFluentCallWithDirective()
+        {
+            var srcBuilder = new StringBuilder(@"/*
+* Generated by the Deep-book trainer application on 5/12/2019 9:31:04 PM
+* Version compatible to 1.0.4.0
+*/
+using Microsoft.ML;
+using Microsoft.ML.Data;
+using Microsoft.ML.Trainers.LightGbm;
+using Microsoft.ML.Transforms;
+
+namespace Cats.Deepbook.Pipeline
+{
+    public static class PipeLineFactory
+    {
+        const string KeyColumn=""KeyColumn"";
+        const string Label=""Label"";
+        const string PredictedLabelIndex=""PredictedLabelIndex"";
+        
+        
+        
+        EstimatorChain <ColumnCopyingTransformer> GenerateDataPipeline(MLContext mlContext, LightGbmMulticlassTrainer.Options options)
+        {
+           return  mlContext.Transforms.Conversion.MapValueToKey(outputColumnName: KeyColumn, inputColumnName: Label)");
```

Comment:
>; [](start = 119, length = 1)

Did you intend this semicolon to be after the whole fluent call chain?  If this is really a test case for the bug, it might be worth doing another test with valid syntax (i.e. move the semicolon from the middle of the fluent call to the end).